meta-mbl-distro/scripts/create-partition-table-markdown

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import pathlib
+import sys
+
+# Set the PYTHONPATH so we can import openembedded scripts/lib modules.
+oe_scripts_path = (
+    pathlib.Path(__file__).resolve().parent.parent.parent.parent
+    / "openembedded-core"
+    / "scripts"
+)
+
+lib_path = oe_scripts_path / "lib"
+sys.path.insert(0, str(lib_path))
+
+import scriptpath
+import scriptutils
+
+# This is needed so we can import bitbake modules.
+bitbakepath = scriptpath.add_bitbake_lib_path()
+
+import bb.tinfoil
+
+import decimal
+
+SCRIPT_NAME = pathlib.Path(__file__).name
+
+def get_var_for_part(d, var_name, part_name, default=None):
+    """
+    Get the value of a BitBake variable for a partition.
+
+    The per-partition variables that MBL defines have names of the form
+    MBL_<part_name>_<var_name>. This function gets the value of one of these
+    variables.
+
+    Args:
+    * d: a BitBake datastore.
+    * var_name: the name of the property to get.
+    * part_name: the name of the partition for which to get the property.
+    * default: the value returned when the requested variable is not set.
+    """
+    full_var_name = "MBL_{}_{}".format(part_name, var_name)
+    val_str = d.getVar(full_var_name, True)
+    if val_str is None and default is not None:
+        return default
+    return val_str
+
+
+def format_size(size_KiB):
+    """
+    Format a size/offset for display.
+
+    Args:
+    * size_KiB: the size/offset in KiB.
+    """
+    # If the size isn't a multiple of 0.5MiB, print it in KiB
+    if size_KiB % 512:
+        return "{}KiB".format(size_KiB)
+    # If the size is a multiple of 1MiB, print it in MiB
+    if not size_KiB % 1024:
+        return "{}MiB".format(size_KiB // 1024)
+    # Otherwise, print it in MiB with one decimal place. The digit after the
+    # decimal point should always be "5", otherwise one of the two earlier
+    # cases should have been used.
+    dec_size_MiB = decimal.Decimal(size_KiB) / 1024
+    one_decimal_place = decimal.Decimal(10) ** -1
+    return "{}MiB".format(dec_size_MiB.quantize(one_decimal_place))
```

Comment:
Could you not just use the precision format specifier instead of converting the size to a `Decimal` object? Something like `return "{:.1f}.format(size_MiB)"`

Reply:
Yes. Don't ask me what I was thinking when I wrote this...

Reply:
https://github.com/ARMmbed/meta-mbl/pull/698/commits/2968ed0dfa47259fe02438569650379f7fa15947